lib/logstash/dead_letter_post_office.rb

```diff
@@ -0,0 +1,53 @@
+# encoding: utf-8
+
+class LogStash::DeadLetterPostOffice
+
+  def self.logger=(logger)
+    @logger = logger
+  end
+
+  def self.logger
+    @logger ||= Cabin::Channel.get(LogStash)
+  end
+
+  def self.destination=(destination)
+    logger.info("Setting dead letter path", :path => destination.location)
+    @destination = destination
+  end
+
+  def self.post(event)
+    Thread.exclusive do
```

Comment:
not really, this is a single purpose synchronisation point, where all threads access it in the same manner buy doing a post on a singleton, so I didn't see a reason to create a specific instance, but I can be convinced otherwise.